pytorch_lightning/trainer/trainer.py

```diff
@@ -626,8 +626,7 @@ def _determine_data_use_amount(self, overfit_batches: float) -> None:
         """Use less data for debugging purposes."""
         if overfit_batches > 0:
             self.limit_train_batches = overfit_batches
-            self.limit_val_batches = overfit_batches
```

Comment:
Does this mean the following?

```python
trainer = Trainer(overfit_batches=3)
trainer.fit(model)  # works as before
trainer.validate()  # now does nothing
trainer.test()  # now does nothing
```

If that's the case, I think the later calls should raise a warning as this is a huge potential footgun. And another pro for #10444

Reply:
this mean the following
`
trainer = Trainer(overfit_batches=3)
trainer.fit(model)  # works as before
trainer.validate()  # now does nothing
trainer.test()  # works as the same with a Trainer without setting `overfit_batches=3` 
`

Reply:
yep, that's a limitation we get for setting these flags inside the trainer.

Reply:
we can rework this in another issue so that overfit batches only applies to the fit loop.

Reply:
I agree it would be expected as it is right now